### PR TITLE
Make sure we always start in a microtick

### PR DESCRIPTION
**Pull request by mcollina**
Signed-off-by: Matteo Collina <hello@matteocollina.com>
Fixes: https://github.com/fastify/fastify/issues/4125

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
lgtm

File: test/await-use.test.js [hunk 1]
```diff
@@ -270,3 +270,25 @@ test('await use - mix of same and future tick callbacks', async (t) => {
   record += 'ready'
   t.equal(record, 'plugin0|plugin1|plugin2|ready')
 })
+
+test('await use - fork the promise chain', { only: true }, (t) => {
```
Comment (stefee) [inline test/await-use.test.js:274]: What does only:true mean?

Comment (delvedor) [reply test/await-use.test.js:274]: It should be removed, it tells to tap to **only** run this test.

Comment (mcollina) [reply test/await-use.test.js:274]: Nothing. If it's run with TAP_ONLY=true env variable, it would run only this test. I'll remove, good spot!

**Pull request merged**